File: Janken.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main {
  public static void main(String[] args) {
    String[] ansList = new String[3];

    ansList[0] = "グー";
    ansList[1] = "チョキ";
    ansList[2] = "パー";

    System.out.println("最初はグー、じゃんけん・・・");
    InputStreamReader isr = new InputStreamReader(System.in);
    BufferedReader br = new BufferedReader(isr);
    System.out.print("あなた：");
    String You = null;
    try {
      You = br.readLine();
      br.close();
    } catch (IOException e) {
      e.printStackTrace();
    };
    int num = (int) (Math.random() * (ansList.length));
    String Java = ansList[num];
    System.out.println("あいて：" + Java);
    if (You.equals(Java)) System.out.println("あいこです");
    else if (You.equals("グー") && Java.equals("チョキ")) System.out.println("あなたの勝ちです");
    else if (You.equals("グー") && Java.equals("パー")) System.out.println("あなたの負けです");
    else if (You.equals("チョキ") && Java.equals("グー")) System.out.println("あなたの負けです");
    else if (You.equals("チョキ") && Java.equals("パー")) System.out.println("あなたの勝ちです");
    else if (You.equals("パー") && Java.equals("グー")) System.out.println("あなたの勝ちです");
    else if (You.equals("パー") && Java.equals("チョキ")) System.out.println("あなたの負けです");
    else System.out.println("正しいポーズを選んでください");

  }
}

File: Janken.exs
ansList = { "グー", "チョキ", "パー" }

IO.puts "最初はグー、じゃんけん・・・"
you = IO.gets "あなた："
num = 0..2 |> Enum.random
elixir = elem(ansList, num)
IO.puts "あいて：#{elixir}";

cond do 
  you == elixir -> IO.puts "あいこです"
  you == "グー\n" && elixir == "チョキ" -> IO.puts "あなたの勝ちです"
  you == "グー\n" && elixir == "パー" -> IO.puts "あなたの負けです"
  you == "チョキ\n" && elixir == "グー" -> IO.puts "あなたの負けです"
  you == "チョキ\n" && elixir == "パー" -> IO.puts "あなたの勝ちです"
  you == "パー\n" && elixir == "グー" -> IO.puts "あなたの勝ちです"
  you == "パー\n" && elixir == "チョキ" -> IO.puts "あなたの負けです"
  true -> IO.puts "正しいポーズを選んでください"
end
